### ISSUE TITLE
Questions about nucleus sampling commit

### ISSUE THREAD

**Issue by sarahwie** [openai/gpt-2#187]
Why is nucleus sampling implemented on top of the _top-k_ logits? (`lines 65-66` in `sample.py`) My understanding from the paper is that the threshold should be implemented on all of the logits.

**Comment by sarahwie** [comment]
I also encounter a bug when using the nucleus sampling implementation stemming from line 36 in `sample.py`. The `tf.where` operation fails due to a dimension mismatch; it seems that `min_values` needs to be expanded with an additional dimension with `vocab_size` elements in order to match the shape of `logits`.
